core/src/main/java/com/github/jsonldjava/utils/JsonUtils.java

```diff
@@ -123,6 +123,16 @@ public static Object fromReader(Reader reader) throws IOException {
             throw new JsonParseException("document doesn't start with a valid json element : "
                     + initialToken, jp.getCurrentLocation());
         }
+        
+        JsonToken t ;
+        try { t = jp.nextToken(); }
```

Comment:
I am a bit tired now but this is confusing me as it seems to be the opposite to what I would expect. What is the rationale behind this throwing an exception if there is still content available, and being silent otherwise. Is it something specific to the use of readValueAs above?

Also, in what situations would it return a value that we are testing for in the if statement after this block?